modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationTemplateLocalServiceImpl.java

```diff
@@ -252,6 +261,16 @@ public NotificationTemplate updateNotificationTemplate(
 			NotificationContext notificationContext)
 		throws PortalException {
 
+		NotificationTemplate notificationTemplate =
+			notificationContext.getNotificationTemplate();
+
+		notificationTemplate = notificationTemplatePersistence.findByPrimaryKey(
+			notificationTemplate.getNotificationTemplateId());
+
+		NotificationTemplateUtil.validateInvokerBundle(
+			"Only allowed bundles can update system notification templates",
+			notificationTemplate.isSystem());
```

Comment:
Because we need to check if the notificationTemplate that we are trying to update is system and the invokerBundles is allowed.
The user can try to update a system nofitication template to a custom notification template, but it shouldn't be possible

Reply:
My question is why do we need to do this in the update method:

1. notificationContext.getNotificationTemplate();
2. notificationTemplatePersistence.findByPrimaryKey(notificationTemplate.getNotificationTemplateId());
3. NotificationTemplateUtil.validateInvokerBundle();

If you see the add method is:

1. notificationContext.getNotificationTemplate();
2. NotificationTemplateUtil.validateInvokerBundle();